torch/cuda/amp/autocast_mode.py

```diff
@@ -57,93 +49,27 @@ def __call__(self, func):
-        )
-        return value.to(dtype) if is_eligible else value
-    elif isinstance(value, (str, bytes)):
-        return value
-    elif HAS_NUMPY and isinstance(value, np.ndarray):
-        return value
-    elif isinstance(value, collections.abc.Mapping):
-        return {_cast(k, dtype): _cast(v, dtype) for k, v in value.items()}
-    elif isinstance(value, collections.abc.Iterable):
-        iterable = (_cast(v, dtype) for v in value)
-        if isinstance(value, (list, tuple)):
-            return type(value)(iterable)
-        else:
-            return iterable
-    else:
-        return value
-
-
-# custom_fwd is a decorator that may or may not be used with arguments, following
-# https://github.com/dabeaz/python-cookbook/tree/master/src/9/defining_a_decorator_that_takes_an_optional_argument.
-# this works:
-#     @custom_fwd
-#     def forward(...):
-# this also works:
-#     @custom_fwd(cast_inputs=torch.float)
-#     def forward(...):
 def custom_fwd(fwd=None, *, cast_inputs=None):
     """
-    Create a helper decorator for ``forward`` methods of custom autograd functions.
-
-    Autograd functions are subclasses of :class:`torch.autograd.Function`.
-    See the :ref:`example page<amp-custom-examples>` for more detail.
-
-    Args:
-        cast_inputs (:class:`torch.dtype` or None, optional, default=None):  If not ``None``,
-            when ``forward`` runs in an autocast-enabled region, casts incoming
-            floating-point CUDA Tensors to the target dtype (non-floating-point Tensors are not affected),
-            then executes ``forward`` with autocast disabled.
-            If ``None``, ``forward``'s internal ops execute with the current autocast state.
-
-    .. note::
-        If the decorated ``forward`` is called outside an autocast-enabled region,
-        :func:`custom_fwd<custom_fwd>` is a no-op and ``cast_inputs`` has no effect.
+    ``torch.cuda.amp.custom_fwd(args...)`` is deprecated. Please use
+    ``torch.amp.custom_fwd(args..., device_type='cuda')`` instead.
     """
-    if fwd is None:
-        return functools.partial(custom_fwd, cast_inputs=cast_inputs)
-
-    @functools.wraps(fwd)
-    def decorate_fwd(*args, **kwargs):
-        args[0]._dtype = torch.get_autocast_gpu_dtype()
-        if cast_inputs is None:
-            args[0]._fwd_used_autocast = torch.is_autocast_enabled()
-            return fwd(*args, **kwargs)
-        else:
-            autocast_context = torch.is_autocast_enabled()
-            args[0]._fwd_used_autocast = False
-            if autocast_context:
-                with autocast(enabled=False):
-                    return fwd(*_cast(args, cast_inputs), **_cast(kwargs, cast_inputs))
-            else:
-                return fwd(*args, **kwargs)
-
-    return decorate_fwd
+    warnings.warn(
+        "torch.cuda.amp.custom_fwd(args...) is deprecated. Please use torch.amp.custom_fwd(args..., device_type='cuda') instead.",
+        DeprecationWarning,
```

Comment:
Since DeprecationWarning are silenced by default and we do want users to see these, we usually use the default warning type for deprecation throughout the codebase.
Feel free to update this and other PRs to not pass this argument (and update test).

Reply:
Got it, thanks.